### PR TITLE
CLN: Make non-empty **kwargs in Index.__new__ fail instead of silently dropped

### PR DESCRIPTION
**Pull request by topper-123**
In #29624 I made change so random kwargs weren't passed to ``Index._simple_new``. I'v found out after merging that the selected solution dropped non-empty kwargs, which isn't great either.

This version ensures that passing non-accepted kwargs raises, so ``pd.Index(['a'], foo="foo")`` raises a TypeError. Instead of raising a TypeError, should we deprecate adding non-accepted kwargs?

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -4259,7 +4266,8 @@ def _concat_same_dtype(self, to_concat, name):
         Concatenate to_concat which has the same class.
         """
         # must be overridden in specific classes
-        klasses = (ABCDatetimeIndex, ABCTimedeltaIndex, ABCPeriodIndex, ExtensionArray)
+        klasses = (ABCDatetimeIndex, ABCTimedeltaIndex, ABCPeriodIndex, ExtensionArray,
+                   ABCIntervalIndex)
```
Comment (topper-123) [inline pandas/core/indexes/base.py]: Without this, if the first element is a IntervalIndex, a"side" parameter is passed to ``_shallow_copy_with_infer`` ind the call will fail.

File: pandas/tests/indexes/multi/test_constructor.py [hunk 2]
```diff
@@ -610,11 +610,10 @@ def test_create_index_existing_name(idx):
             ],
             dtype="object",
         ),
-        names=["foo", "bar"],
     )
     tm.assert_index_equal(result, expected)
 
-    result = pd.Index(index, names=["A", "B"])
+    result = pd.Index(index, name="A")
```
Comment (topper-123) [inline pandas/tests/indexes/multi/test_constructor.py]: Both result an dexpected are normal ``Index`` and instantiating with a ``names`` parameter should fail.

File: pandas/core/indexes/base.py [hunk 3]
```diff
@@ -450,7 +451,13 @@ def __new__(
                             return PeriodIndex(subarr, name=name, **kwargs)
                         except IncompatibleFrequency:
                             pass
-            return cls._simple_new(subarr, name)
+            if kwargs:
+                if len(kwargs) == 1:
+                    msg = f"Unexpected keyword argument {list(kwargs)[0]!r}"
+                else:
+                    msg = f"Unexpected keyword arguments {set(kwargs)!r}"
```
Comment (jreback) [inline pandas/core/indexes/base.py]: I would just use this one (the set), no need to make a super-detailed error message

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. merge on green.

**Pull request merged**